### ISSUE TITLE
Feb Extension o²S²PARC Meta-Modeling Framework: Propagate Probability Distributions

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#1110]
Description:
Extension of the o²S²PARC meta-modeling framework to allow propagating (multi-dimensional) parameter probability distributions and sets of data samples through a study pipeline and extract statistical characteristics about computed quantities of interest.

Deliverable:
o²S²PARC functionality and demonstrator

Acceptance criteria: 
A demonstrator exists that illustrates propagation of probability distributions.

Out of scope: -

Deadline:
Q2

[wrike Y7-MS 11.1.1.1: Extension o²S²PARC Meta-Modeling Framework: Propagate Probability Distributions](https://www.wrike.com/open.htm?id=1162027388)

**Comment by elisabettai** [comment]
Esra should talk with Werner and Antonino. Werner to organize meeting.

**Comment by elisabettai** [comment]
Y7Q2 Status update (02/29/2024): Completed
Implementation: A demonstration of this functionality was presented at the monthly update meeting on 02/28/2024. 
A new Map Service allows to run study clones in parallel over a set of parameters and returns the probed results (study output ports).
A service based on [Dakota](https://dakota.sandia.gov/) was implemented that leverage Dakota’s powerful uncertainty propagation functionality. The Dakota algorithms accept probability distributions of the input variables, (data derived) histograms, and (fitted) analytical distributions. They sample distributions smartly (sparsely, but with maximized information content), calculate the resulting model responses and compute probability distributions of the output quantities. The Dakota service uses a new Map service to run the model for several parameter sets in parallel. Dakota can also build surrogate models to further minimize the number of required (expensive) samples and improve the response surface fit. 
The uncertainty propagation implementation was validated using benchmarks with known analytical solutions.
